test/cql-pytest/test_materialized_view.py

```diff
@@ -992,3 +992,46 @@ def test_base_clustering_prefix_deletion(cql, test_keyspace):
             # After the deletion, all data should be gone from both base and view
             assert [] == list(cql.execute(f"SELECT c2 FROM {table}"))
             assert [] == list(cql.execute(f"SELECT c2 FROM {mv}"))
+
+def test_base_partition_deletion_with_same_partition_key(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'p int, c int, v int, primary key (p,c)') as table:
+        # All inserts go to the same base partition,  that we'll then delete
+        v = 42
+        insert = cql.prepare(f'INSERT INTO {table} (p,c,v) VALUES (?,?,{v})')
+        # Case where all view-row deletions go to the same view partition:
+        with (
+            new_materialized_view(cql, table, '*', 'p,c', 'p is not null and c is not null') as mv1,
+            new_materialized_view(cql, table, '*', 'c,p', 'p is not null and c is not null') as mv2,
+            new_materialized_view(cql, table, '*', '(p,c),v', 'p is not null and c is not null and v is not null') as mv3,
+        ):
+            N = 10
+            for i in range(N):
+                cql.execute(insert, [1, i])
+                cql.execute(insert, [2, i])
+
+            # Before the deletion, all N rows should exist in the base and the
+            # view
+            allN = list(range(N))
+            assert allN == [x.c for x in cql.execute(f"SELECT c FROM {table} WHERE p=1")]
```

Comment:
Materialized view updates are asynchronous, so the assertions you added might be flaky. You can add `WITH synchronous_updates = true` at the end of the `CREATE MATERIALIZED VIEWS` statements to change that behavior.